### ISSUE TITLE
Deformation of objects of the same type

### ISSUE THREAD

**Issue by huassaue** [rqhuang88/DV-Matcher#12]
<img width="762" height="195" alt="Image" src="https://github.com/user-attachments/assets/dce62eff-96f1-4a33-a6c4-7efeca9986b0" />
I trained several different cup models for 20 epochs (normalized and sampled to 2000 points). The images from left to right are source, target, pi-verts, and deform. Why can the initial result be deformed to approach the target, but the final deformation returns to the source？

**Comment by huassaue** [comment]
And may I ask what the approximate value of loss is during convergence

**Comment by huassaue** [comment]
<img width="872" height="237" alt="Image" src="https://github.com/user-attachments/assets/60f86dc7-9bca-4e68-ad02-62e732fa727f" /> 部分点云变形配准时，要获得自己数据的部分点云，可以使用注释掉的这些代码吗？

**Comment by zhangquanchen** [comment]
> <img alt="Image" width="872" height="237" src="https://private-user-images.githubusercontent.com/107062689/489417474-60f86dc7-9bca-4e68-ad02-62e732fa727f.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTgwOTc1NTksIm5iZiI6MTc1ODA5NzI1OSwicGF0aCI6Ii8xMDcwNjI2ODkvNDg5NDE3NDc0LTYwZjg2ZGM3LTliY2EtNGU2OC1hZDAyLTYyZTczMmZhNzI3Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkxN1QwODIwNTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kMmMyZmMwY2JmNzMyMDYwOGU1YzA4N2FmMDY0YzNmNmIxZWQ5YzZmN2UzYTBhZWIxNDY0ZmI4Yjk0ZTlmOWFjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.R3_caNHK9PSLDJ3Y1BjZlNeY-t6LdEQX0rJUGnNtzu8"> 部分点云变形配准时，要获得自己数据的部分点云，可以使用注释掉的这些代码吗？

Of course, our partial-view data is also constructed based on these codes, from [here](https://github.com/rqhuang88/DV-Matcher/issues/12#issuecomment-3290921590).

**Comment by zhangquanchen** [comment]
> And may I ask what the approximate value of loss is during convergence

Thanks for your interest. Our converged training loss on SCAPE is about 1.5 (total). But, different setting may have some differences.

**Comment by zhangquanchen** [comment]
@huassaue 

Actually, we have not validated our method on certain furniture (such as cups, chairs) yet. However, we think this could be a very **interesting follow-up**, and we appreciate and welcome you to try it.

However, there might be many adjustments needed, which could be the reasons causing the issue you mentioned:
1. We are not sure whether the style variations of cups would exceed the capabilities of the existing deformer based on **local rigid transformations**.
2. You might need to reconsider some hyperparameter designs for the loss function during the training process. Due to the significant deformation, **the corresponding weights should change**.
3. Some regularization terms or stronger **spatial** constraints might be needed.

**Comment by huassaue** [comment]
> [@huassaue](https://github.com/huassaue)
> 
> 实际上，我们尚未在某些家具（例如杯子、椅子）上验证我们的方法。不过，我们认为这可能是一个非常**有趣的后续研究**，我们非常欢迎并感谢你尝试。
> 
> 但是，可能需要进行很多调整，这可能是导致您提到的问题的原因：
> 
> 1. 我们不确定杯子的样式变化是否会超出基于**局部刚性变换**的现有变形器的能力。
> 2. 在训练过程中，你可能需要重新考虑损失函数的一些超参数设计。由于显著的变形，**相应的权重应该会发生变化**。

我已经做到了整体点云的变形配准（杯子等），现在进行尝试partial（杯子等），仍然会出现图片中的情况

**Comment by huassaue** [comment]
> [@huassaue](https://github.com/huassaue)
> 
> 实际上，我们尚未在某些家具（例如杯子、椅子）上验证我们的方法。不过，我们认为这可能是一个非常**有趣的后续研究**，我们非常欢迎并感谢你尝试。
> 
> 但是，可能需要进行很多调整，这可能是导致您提到的问题的原因：
> 
> 1. 我们不确定杯子的样式变化是否会超出基于**局部刚性变换**的现有变形器的能力。
> 2. 在训练过程中，你可能需要重新考虑损失函数的一些超参数设计。由于显著的变形，**相应的权重应该会发生变化**。

我已经实现了整体点云的变形配准（杯子等），现在正在进行尝试部分点云配准（杯子等），仍然会出现图片中的情况，pi-verts 结果可以观察到向target变形，但最终模型预测的变形结果和source相近（似乎有什么参数影响他不进行变形/配准）

**Comment by huassaue** [comment]
> > [@huassaue](https://github.com/huassaue)
> > 实际上，我们尚未在某些家具（例如杯子、椅子）上验证我们的方法。不过，我们认为这可能是一个非常**有趣的后续研究**，我们非常欢迎并感谢你尝试。
> > 但是，可能需要进行很多调整，这可能是导致您提到的问题的原因：
> > 
> > 1. 我们不确定杯子的样式变化是否会超出基于**局部刚性变换**的现有变形器的能力。
> > 2. 在训练过程中，你可能需要重新考虑损失函数的一些超参数设计。由于显著的变形，**相应的权重应该会发生变化**。
> 
> 我已经实现了整体点云的变形配准（杯子等），现在正在进行尝试部分点云配准（杯子等），仍然会出现图片中的情况，pi-verts 结果可以观察到向target变形，但最终模型预测的变形结果和source相近（似乎有什么参数影响他不进行变形/配准）

@zhangquanchen

**Comment by zhangquanchen** [comment]
> > [@huassaue](https://github.com/huassaue)
> > 实际上，我们尚未在某些家具（例如杯子、椅子）上验证我们的方法。不过，我们认为这可能是一个非常**有趣的后续研究**，我们非常欢迎并感谢你尝试。
> > 但是，可能需要进行很多调整，这可能是导致您提到的问题的原因：
> > 
> > 1. 我们不确定杯子的样式变化是否会超出基于**局部刚性变换**的现有变形器的能力。
> > 2. 在训练过程中，你可能需要重新考虑损失函数的一些超参数设计。由于显著的变形，**相应的权重应该会发生变化**。
> 
> 我已经实现了整体点云的变形配准（杯子等），现在正在进行尝试部分点云配准（杯子等），仍然会出现图片中的情况，pi-verts 结果可以观察到向target变形，但最终模型预测的变形结果和source相近（似乎有什么参数影响他不进行变形/配准）

The **deformation weight** during the training process is crucial

**Comment by zhangquanchen** [comment]
@huassaue 
You can try to **reduce ARAP** and **change the number of nodes in the deformation graph**. I think this might help

**Comment by huassaue** [comment]
> > > [@huassaue](https://github.com/huassaue)
> > > 实际上，我们尚未在某些家具（例如杯子、椅子）上验证我们的方法。不过，我们认为这可能是一个非常**有趣的后续研究**，我们非常欢迎并感谢你尝试。
> > > 但是，可能需要进行很多调整，这可能是导致您提到的问题的原因：
> > > 
> > > 1. 我们不确定杯子的样式变化是否会超出基于**局部刚性变换**的现有变形器的能力。
> > > 2. 在训练过程中，你可能需要重新考虑损失函数的一些超参数设计。由于显著的变形，**相应的权重应该会发生变化**。
> > 
> > 
> > 我已经实现了整体点云的变形配准（杯子等），现在正在进行尝试部分点云配准（杯子等），仍然会出现图片中的情况，pi-verts 结果可以观察到向target变形，但最终模型预测的变形结果和source相近（似乎有什么参数影响他不进行变形/配准）
> 
> The **deformation weight** during the training process is crucial

@zhangquanchen 

<img width="216" height="182" alt="Image" src="https://github.com/user-attachments/assets/f8bdbd6d-6ea0-4576-95bd-bbfc2a80603b" /> 具体是哪个呢

**Comment by zhangquanchen** [comment]
@huassaue It is w_arap. ARAP would make the deformation as rigid as possible, but for cup, the deformation somehow should be bigger.
Besides, the number of nodes in the deformation graph can be found in [lib](https://github.com/rqhuang88/DV-Matcher/tree/main/lib/deformation_graph_point.py).

**Comment by huassaue** [comment]
> [@huassaue](https://github.com/huassaue)答案是 w_arap。ARAP 会让变形尽可能地刚性，但对于杯子来说，变形应该更大一些。 此外，变形图中的节点数可以在[lib](https://github.com/rqhuang88/DV-Matcher/tree/main/lib/deformation_graph_point.py)中找到。

您觉得节点数应该增多还是减少呢 感谢 @zhangquanchen

**Comment by zhangquanchen** [comment]
> > [@huassaue](https://github.com/huassaue)答案是 w_arap。ARAP 会让变形尽可能地刚性，但对于杯子来说，变形应该更大一些。 此外，变形图中的节点数可以在[lib](https://github.com/rqhuang88/DV-Matcher/tree/main/lib/deformation_graph_point.py)中找到。
> 
> 您觉得节点数应该增多还是减少呢 感谢 [@zhangquanchen](https://github.com/zhangquanchen)

my opinion is less, thanks
